src/motion_input_shaping/shaped_lockstep.py

```diff
@@ -0,0 +1,549 @@
+"""
```

Comment:
I think Jay is doing this specifically because he wanted to include a gantry example in a future article. I agree there is a ton of duplicated code though and it would be nice if we could combine it. That being said, having this file doesn't hurt. "Performs the same thing except for lockstep" is pretty intuitive.

Reply:
This is now combined with `shaped_axis.py`